scheduler/src/cook/mesos/data_locality.clj

```diff
@@ -1,32 +1,118 @@
+          clean-job->host->cost (pc/map-vals (fn [host->cost]
+                                               (pc/map-vals (fn [cost] (-> cost (min maximum-cost) (max 0)))
+                                                            host->cost))
+                                             job->host->cost)]
+      (swap! data-local-costs (fn update-data-local-costs-atom
+                                [current]
+                                (let [remove-old-values (apply dissoc current uuids-to-remove)]
+                                  (merge remove-old-values clean-job->host->cost))))
+      (swap! last-update-time (fn update-last-update-time
+                                [current]
+                                (let [remove-old-values (apply dissoc current uuids-to-remove)
+                                      new-values (pc/map-from-keys (constantly (t/now)) (keys clean-job->host->cost))]
+                                  (merge remove-old-values new-values))))))
 
   (defn get-data-local-costs
     "Returns the current cost for jobs to run on each host"
     []
-    @data-local-costs))
+    @data-local-costs)
+
+  (defn job-ids-to-update
+    "Return the list of job ids to update data locality costs for in the next iteration.
+     The number of jobs is limited by the configured batch size."
+    [db]
+    (let [{:keys [batch-size]} (config/data-local-fitness-config)
+          pending-jobs (->> db
+                            util/get-pending-job-ents
+                            (filter :job/data-local))
+          last-update-time-val @last-update-time
+          have-data?->job (group-by (fn [job] (contains? last-update-time-val (:job/uuid job)))
+                                    pending-jobs)
+          missing-data (->> (get have-data?->job false [])
+                            (sort-by :job/submit-time)
+                            (map :job/uuid))
+          sorted-have-data (->> (get have-data?->job true [])
+                                (map :job/uuid)
+                                (sort-by last-update-time-val))
+          pending-job-uuids (->> pending-jobs
+                                 (map :job/uuid)
+                                 (into (hash-set)))
+          no-longer-waiting (set/difference (into (hash-set) (keys last-update-time-val))
+                                            pending-job-uuids)]
+      {:to-fetch (into [] (take batch-size (concat missing-data sorted-have-data)))
+       :to-remove no-longer-waiting})))
+
+(defn fetch-data-local-costs
+  "Contacts the server to obtain the data local costs for the given job ids"
+  [job-ids]
+  (let [{:keys [cost-endpoint]} (config/data-local-fitness-config)
+        request {:batch (UUID/randomUUID)
+                 :tasks job-ids}
+        _ (log/debug "Updating data local costs for: " job-ids)
+        {:keys [body]} (http/post cost-endpoint {:body (cheshire/generate-string request)
```

Comment:
We should have a unit tests that acts as documentation for the expected POST body and response template.